homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -145,7 +154,7 @@ class EnvoyProductionSensorEntityDescription(
 
 
 PRODUCTION_PHASE_SENSORS = {
-    (on_phase := PhaseNames(PHASENAMES[phase])): [
+    (on_phase := str(PhaseNames(PHASENAMES[phase]))): [
```

Comment:
```suggestion
    (on_phase := PHASENAMES[phase]): [
```

Does it need to be converted to an enum and than back to a `str`?

Reply:
When using PHASENAMES[phase] linter starts complaining in line 561: Argument of type "str" cannot be assigned to parameter "__key" of type "PhaseNames" in function "__getitem__"
  "str" is incompatible with "PhaseNames"

Reply:
Updated suggestion

Reply:
I think mypy can't work out the const

`PHASENAMES: list[PhaseNames] = list(PhaseNames)`

Probably needs a hint

Reply:
```diff
diff --git a/homeassistant/components/enphase_envoy/sensor.py b/homeassistant/components/enphase_envoy/sensor.py
index 8b19be47b76..7d315de0fdb 100644
--- a/homeassistant/components/enphase_envoy/sensor.py
+++ b/homeassistant/components/enphase_envoy/sensor.py
@@ -16,7 +16,7 @@ from pyenphase import (
     EnvoySystemConsumption,
     EnvoySystemProduction,
 )
-from pyenphase.const import PHASENAMES, PhaseNames
+from pyenphase.const import PHASENAMES
 from pyenphase.models.meters import (
     CtMeterStatus,
     CtState,
@@ -154,7 +154,7 @@ PRODUCTION_SENSORS = (
 
 
 PRODUCTION_PHASE_SENSORS = {
-    (on_phase := str(PhaseNames(PHASENAMES[phase]))): [
+    (on_phase := PHASENAMES[phase].value): [
         replace(
             sensor,
             key=f"{sensor.key}_l{phase + 1}",
@@ -231,7 +231,7 @@ CONSUMPTION_SENSORS = (
 
 
 CONSUMPTION_PHASE_SENSORS = {
-    (on_phase := str(PhaseNames(PHASENAMES[phase]))): [
+    (on_phase := PHASENAMES[phase].value): [
         replace(
             sensor,
             key=f"{sensor.key}_l{phase + 1}",
@@ -340,7 +340,7 @@ CT_NET_CONSUMPTION_SENSORS = (
 
 
 CT_NET_CONSUMPTION_PHASE_SENSORS = {
-    (on_phase := str(PhaseNames(PHASENAMES[phase]))): [
+    (on_phase := PHASENAMES[phase].value): [
         replace(
             sensor,
             key=f"{sensor.key}_l{phase + 1}",
@@ -374,7 +374,7 @@ CT_PRODUCTION_SENSORS = (
 )
 
 CT_PRODUCTION_PHASE_SENSORS = {
-    (on_phase := str(PhaseNames(PHASENAMES[phase]))): [
+    (on_phase := PHASENAMES[phase].value): [
         replace(
             sensor,
             key=f"{sensor.key}_l{phase + 1}",

```

Reply:
This works just fine, thanks

Reply:
And then mypy came back: homeassistant/components/enphase_envoy/sensor.py:157: error: "str" has no attribute "value"  [attr-defined]

Reply:
Added type hint to pyenphase. For now added same local definition for PHASENAMES until pyenphase is bumped.

Reply:
Removed local PHASENAMES and import from updated pyenphase again.